user-interface/src/main/java/life/qbic/datamanager/views/register/UserRegistrationLayout.java

```diff
@@ -55,19 +72,76 @@ public class UserRegistrationLayout extends VerticalLayout {
   private VerticalLayout fieldLayout;
   private final VerticalLayout contentLayout;
   private H2 layoutTitle;
+  private IdentityService identityService;
 
-  public UserRegistrationLayout(@Autowired UserRegistrationHandlerInterface registerHandler) {
-
+  public UserRegistrationLayout(@Autowired IdentityService identityService) {
+    this.identityService = Objects.requireNonNull(identityService);
     this.addClassName("grid");
-    contentLayout = new VerticalLayout();
-
+    this.contentLayout = new VerticalLayout();
     initLayout();
     styleLayout();
-    registerToHandler(registerHandler);
+    initFields();
+    addListener();
+  }
+
+  private void initFields() {
+    fullName.setPattern("\\S.*");
+    fullName.setErrorMessage("Please provide your full name here");
+    email.setErrorMessage("Please provide a valid mail address");
+    password.setHelperText("A password must be at least 12 characters");
+    password.setPattern(".{12,}");
+    password.setErrorMessage("Password too short");
+    username.setHelperText("Your unique username, visible to other users");
+    username.setErrorMessage("Please provide a username");
   }
 
-  private void registerToHandler(UserRegistrationHandlerInterface registerHandler) {
-    registerHandler.handle(this);
+  private void addListener() {
+    registerButton.addClickShortcut(Key.ENTER);
+
+    registerButton.addClickListener(
+        event -> {
+          clearNotifications();
+          var response = identityService.registerUser(
+              fullName.getValue().strip(),
+              username.getValue().strip(),
+              email.getValue().strip(),
+              password.getValue().toCharArray()
+             );
+          handleResponse(response);
+        });
+  }
+
+  private void handleResponse(ApplicationResponse response) {
+    response.ifSuccessOrElse(this::onUserRegistrationSucceeded, this::handleRegistrationFailure);
+  }
+
+  private void handleRegistrationFailure(ApplicationResponse response) {
+    UserRegistrationException exception = build(response);
+    handleRegistrationFailure(exception);
+  }
+
+  private UserRegistrationException build(ApplicationResponse applicationResponse) {
```

Comment:
```suggestion
  private UserRegistrationException convertToRegistrationException(ApplicationResponse applicationResponse) {
```
The method name `build` does not give any information in this context.